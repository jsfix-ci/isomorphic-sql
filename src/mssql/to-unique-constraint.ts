export function toUniqueConstraint (table: string, columns: string[]) {
    return `ALTER TABLE ${table} ADD CONSTRAINT IX_${table}_${columns.join('_')}_unique UNIQUE(${columns.join(', ')})`;
};