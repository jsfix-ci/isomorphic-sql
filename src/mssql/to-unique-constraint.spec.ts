import { toUniqueConstraint } from './to-unique-constraint';

describe('toUniqueConstraint', () => {
    describe('single column use', () => {
        it('returns an add-constraint expression for uniqueness', () => {
            expect(toUniqueConstraint('foods', ['plu'])).toMatchSnapshot();
        });
    });

    describe('multi column use', () => {
        it('returns an add-constraint expression for uniqueness', () => {
            expect(toUniqueConstraint('events', ['event_date', 'event_location'])).toMatchSnapshot();
        });
    });
});