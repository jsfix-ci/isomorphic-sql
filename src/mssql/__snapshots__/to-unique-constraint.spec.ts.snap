// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toUniqueConstraint multi column use returns an add-constraint expression for uniqueness 1`] = `"ALTER TABLE events ADD CONSTRAINT IX_events_event_date_event_location_unique UNIQUE(event_date, event_location)"`;

exports[`toUniqueConstraint single column use returns an add-constraint expression for uniqueness 1`] = `"ALTER TABLE foods ADD CONSTRAINT IX_foods_plu_unique UNIQUE(plu)"`;
